bundles/org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/link/TypeConverters.java

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.webthing.internal.link;
+
+import java.awt.*;
+import java.math.BigDecimal;
+import java.util.Collection;
+import java.util.Locale;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.library.types.*;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+
+/**
+ * Helper class to create a TypeConverter
+ *
+ * @author Gregor Roth - Initial contribution
+ */
+@NonNullByDefault
+class TypeConverters {
+
+    /**
+     * create a TypeConverter for a given Item type and property type
+     * 
+     * @param itemType the item type
+     * @param propertyType the property type
+     * @return the type converter
+     */
+    static TypeConverter create(String itemType, String propertyType) {
+        switch (itemType.toLowerCase(Locale.ENGLISH)) {
+            case "switch":
+                return new SwitchTypeConverter();
+            case "dimmer":
+                return new DimmerTypeConverter();
+            case "contact":
+                return new ContactTypeConverter();
+            case "color":
+                return new ColorTypeConverter();
+            case "number":
+                if (propertyType.toLowerCase(Locale.ENGLISH).equals("integer")) {
+                    return new IntegerTypeConverter();
+                } else {
+                    return new NumberTypeConverter();
+                }
+            default:
+                return new StringTypeConverter();
+        }
+    }
+
+    private static boolean toBoolean(Object propertyValue) {
+        return Boolean.parseBoolean(propertyValue.toString());
+    }
+
+    private static BigDecimal toDecimal(Object propertyValue) {
+        return new BigDecimal(propertyValue.toString());
+    }
+
+    private static final class ColorTypeConverter implements TypeConverter {
+
+        @Override
+        public Command toStateCommand(Object propertyValue) {
+            var value = propertyValue.toString();
+            if (!value.contains("#")) {
+                value = "#" + value;
+            }
+            Color rgb = Color.decode(value);
+            float[] hsb = new float[3];
+            hsb = Color.RGBtoHSB(rgb.getRed(), rgb.getGreen(), rgb.getBlue(), hsb);
+            String hsbString = (hsb[0] * 360) + "," + (hsb[1] * 100) + "," + (hsb[2] * 100);
+            return HSBType.valueOf(hsbString);
```

Comment:
Is there a reason why you don't use `HSBType.fromRGB()`?

Reply:
Thanks for the hint. fixed